modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/DataRegion.java

```diff
@@ -80,4 +87,25 @@ public DataRegionMetricsImpl memoryMetrics() {
     public PageEvictionTracker evictionTracker() {
         return evictionTracker;
     }
+
+    /**
+     * Increase the pool of empty pages, if it is too small.
+     *
+     * @return {@code true} if increased.
+     */
+    public synchronized boolean increaseEmptyPagesPool() {
+        if (cfg != null && pageMem != null
+            && emptyPagesPoolSize < (cfg.getMaxSize() / pageMem.systemPageSize() * (1 - cfg.getEvictionThreshold()))) {
+            emptyPagesPoolSize *= 2;
```

Comment:
perhaps, it would be as follows: `emptyPagesPoolSize = max(emptyPagesPoolSize * 2, rightBorder);`